model/src/test/kotlin/AbstractDependencyNavigatorTest.kt

```diff
@@ -0,0 +1,360 @@
+                    ),
+                    Identifier("Maven:com.fasterxml.jackson.core:jackson-core:2.8.9") to listOf(
+                        Identifier("Maven:net.logstash.logback:logstash-logback-encoder:4.11"),
+                        Identifier("Maven:com.fasterxml.jackson.core:jackson-databind:2.8.9")
+                    ),
+                    Identifier("Maven:com.fasterxml.jackson.core:jackson-databind:2.8.9") to listOf(
+                        Identifier("Maven:net.logstash.logback:logstash-logback-encoder:4.11")
+                    ),
+                    Identifier("Maven:com.typesafe.akka:akka-actor_2.12:2.5.6") to listOf(
+                        Identifier("Maven:com.typesafe.akka:akka-stream_2.12:2.5.6")
+                    ),
+                    Identifier("Maven:com.typesafe.akka:akka-stream_2.12:2.5.6") to emptyList(),
+                    Identifier("Maven:com.typesafe:config:1.3.1") to emptyList(),
+                    Identifier("Maven:com.typesafe.scala-logging:scala-logging_2.12:3.7.2") to emptyList(),
+                    Identifier("Maven:com.typesafe:ssl-config-core_2.12:0.2.2") to listOf(
+                        Identifier("Maven:com.typesafe.akka:akka-stream_2.12:2.5.6")
+                    ),
+                    Identifier("Maven:net.logstash.logback:logstash-logback-encoder:4.11") to emptyList(),
+                    Identifier("Maven:org.scala-lang:scala-library:2.12.3") to emptyList(),
+                    Identifier("Maven:org.scala-lang:scala-reflect:2.12.2") to listOf(
+                        Identifier("Maven:com.typesafe.scala-logging:scala-logging_2.12:3.7.2")
+                    ),
+                    Identifier("Maven:org.scala-lang.modules:scala-java8-compat_2.12:0.8.0") to listOf(
+                        Identifier("Maven:com.typesafe.akka:akka-stream_2.12:2.5.6"),
+                        Identifier("Maven:com.typesafe.akka:akka-actor_2.12:2.5.6")
+                    ),
+                    Identifier("Maven:org.reactivestreams:reactive-streams:1.0.1") to listOf(
+                        Identifier("Maven:com.typesafe.akka:akka-stream_2.12:2.5.6")
+                    ),
+                    Identifier("Maven:org.slf4j:jcl-over-slf4j:1.7.25") to emptyList(),
+                    Identifier("Maven:org.slf4j:slf4j-api:1.7.25") to listOf(
+                        Identifier("Maven:ch.qos.logback:logback-classic:1.2.3")
+                    ),
+                )
+            }
+        }
+
+        "projectDependencies" should {
+            "return the dependencies of a project" {
+                val scopeDependencies = navigator.scopeDependencies(testProject)
+                val projectDependencies = navigator.projectDependencies(testProject)
+
+                projectDependencies should containExactlyInAnyOrder(
+                    scopeDependencies["compile"].orEmpty() + scopeDependencies["test"].orEmpty()
```

Comment:
Reminder: Please consider avoiding the implicit non-null checks also here, and in other places using `orEmpty()`.

Reply:
Done, but for this specific place, is there a more elegant solution than nesting `shouldNotBeNull` blocks?

Reply:
I don't think so... maybe @mnonnenmacher has an idea?

Reply:
You could write it like this to avoid nesting:
```
scopeDependencies should containKeys("compile", "test")
val expectedDependencies = scopeDependencies.getValue("compile") + scopeDependencies.getValue("test")
projectDependencies should containExactlyInAnyOrder(compileDependencies + testDependencies)
```